src/bench/mempool_eviction.cpp

```diff
@@ -21,116 +21,94 @@ static void AddTx(const CTransactionRef& tx, const CAmount& nFee, CTxMemPool& po
         spendsCoinbase, sigOpCost, lp));
 }
 
-// Right now this is only testing eviction performance in an extremely small
-// mempool. Code needs to be written to generate a much wider variety of
-// unique transactions for a more meaningful performance measurement.
-static void MempoolEviction(benchmark::Bench& bench)
+void add_parents_child(std::vector<CTransactionRef>& txns, size_t package_size, size_t set_num)
+{
+    // N+1 N-input-N-output txns, in a parents-and-child package
+    // Where each tx takes an input from all prior txns
+    const size_t num_puts = std::max(package_size - 1, (size_t)1);
+    const size_t num_txns = package_size;
+    for (size_t txn_index=0; txn_index<num_txns; txn_index++) {
+        CMutableTransaction tx = CMutableTransaction();
+        tx.vin.resize(num_puts);
+        tx.vout.resize(num_puts);
+        for (size_t put_index=0; put_index<num_puts; put_index++) {
```

Comment:
Can you apply the changes to the other loops too? i.e.
```suggestion
        for (size_t put_index{0}; put_index < num_puts; ++put_index) {
```